src/main/java/org/jboss/forge/arquillian/command/cube/CubeSetupCommand.java

```diff
@@ -0,0 +1,336 @@
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.jboss.forge.arquillian.util.StringUtil.getStringForCLIDisplay;
+
+public class CubeSetupCommand extends AbstractProjectCommand implements UICommand {
+
+    private CubeSetupFacet cubeSetupFacet;
+
+    @Inject
+    private ProjectFactory projectFactory;
+
+    @Inject
+    private FacetFactory facetFactory;
+
+    @Inject
+    private ResourceFactory resourceFactory;
+
+    @Inject
+    @WithAttributes(shortName = 't', label = "Type", type = InputType.DROPDOWN)
+    private UISelectOne<String> type;
+
+    @Inject
+    @WithAttributes(shortName = 'f', label = "File", required = true)
+    private UIInput<String> filePath;
+
+    @Inject
+    @WithAttributes(shortName = 'm', label = "Docker Machine", type = InputType.CHECKBOX)
+    private UIInput<Boolean> dockerMachine;
+
+    @Inject
+    @WithAttributes(shortName = 'n', label = "Docker Machine Name")
+    private UIInput<String> machineName;
+
+    @Inject
+    @WithAttributes(shortName = 'd', label = "Docker File Name")
+    private UIInput<String> dockerFileName;
+
+
+    @Override
+    public void initializeUI(UIBuilder builder) throws Exception {
+        builder
+            .add(type)
+            .add(filePath)
+            .add(dockerMachine)
+            .add(machineName)
+            .add(dockerFileName);
+
+        final List<String> types = Arrays.stream(CubeConfiguration.values())
+            .map(CubeConfiguration::getType)
+            .collect(Collectors.toList());
+
+        type.setValueChoices(types);
+        type.setEnabled(true);
+        type.setRequired(() -> true);
+        type.setItemLabelConverter(source -> {
+            if (source == null) {
+                return null;
+            }
+            if (builder.getUIContext().getProvider().isGUI()) {
+                return source;
+            }
+            return getStringForCLIDisplay(source);
+        });
+
+        filePath.setEnabled(true);
+
+        dockerMachine.setEnabled(this::isDockerOrDockerCompose);
+        dockerMachine.setRequired(this::isDockerOrDockerCompose);
+
+        machineName.setEnabled(this::isDockerMachine);
+        machineName.setRequired(this::isDockerMachine);
+
+        dockerFileName.setEnabled(this::isDocker);
+    }
+
+    @Override
+    public UICommandMetadata getMetadata(UIContext context) {
+        return Metadata.from(super.getMetadata(context), getClass())
+            .category(Categories.create("Arquillian"))
+            .name("Arquillian: Cube Setup")
+            .description("This addon will help you setup Arquillian Cube.");
+    }
+
+    @Override
+    public Result execute(UIExecutionContext context) throws Exception {
+
+        final String msg = checkResourcesExists(context);
+
+        if (type.hasValue()) {
+            setCubeSetupFacet();
+            setCubeConfigurationParamteres(context);
+        }
+
+        if (msg != null) {
```

Comment:
Maybe we should fail before calling `setCubeSetupFacet()` and `setCubeConfigurationParamteres(context)`?